### PR TITLE
Link & Unlink

### PR DESCRIPTION
**Pull request by yoav-lavi**
# Description

## Features

- Implements `link` and `unlink`

## Fixes

- Uses paginated APIs in place of non paginated ones, limits results to 100 (5 for API keys)

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [x] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: cli/crates/backend/src/api/link.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+use cynic::{http::ReqwestExt, QueryBuilder};
+use std::iter;
+
+/// # Errors
+///
+/// see [`ApiError`]
+#[allow(clippy::module_name_repetitions)]
+pub async fn get_viewer_data_for_link() -> Result<Vec<AccountWithProjects>, ApiError> {
+    // TODO consider if we want to do this elsewhere
+    if project_linked() {
```
Comment (jakubadamw) [inline cli/crates/backend/src/api/link.rs:19]: We seem to be mixing synchronous APIs with async. This checks if a file exists with a synchronous API within an `async` function. I know the CLi is a mix match but whenever we're in an `async` scope already, it'd be desirable for all I/O within to play nicely as well. Or we wrap it with `tokio::task::spawn_blocking()` if there's no async API to do the same thing.

Comment (jakubadamw) [reply cli/crates/backend/src/api/link.rs:19]: I guess all I meant to say is that we should use [`tokio::fs::try_exists()`](https://docs.rs/tokio/latest/tokio/fs/fn.try_exists.html) here. 🙂

**Review by jakubadamw — APPROVED** [review]
One issue around I/O API mixing, looks great otherwise! Great work!

File: cli/crates/backend/src/api/link.rs (hunk 1 above)
Comment (yoav-lavi) [reply cli/crates/backend/src/api/link.rs]: Makes sense, fixed!

**Pull request merged**